### PR TITLE
adding asynchronous spinner to manage callbacks across multiple threads

### PR DESCRIPTION
**Pull request by VaidehiSom**

### CODE REVIEW

File: src/rosbag_recorder_node.cpp [hunk 1]
```diff
@@ -56,13 +56,18 @@ int main(int argc, char** argv) {
...
 
+    // Use an asynchronous spinner to manage callbacks across multiple threads
+    ros::AsyncSpinner spinner(4); // Use 4 threads
+    spinner.start();
+
     //main loop
     ros::Rate r(1000);
     while(ros::ok()) {
         //bag_launcher's loop function
         bag_launcher.check_all();
```
Comment (gus-robotics88) [inline src/rosbag_recorder_node.cpp]: Could we get rid of this while loop and instead create a ROS Timer in the constructor to BagLauncher that launches this timer which does a check_all() upon timer events?

Comment (VaidehiSom) [reply src/rosbag_recorder_node.cpp]: Nice idea! Updated

File: src/bag_launcher.cpp [hunk 2]
```diff
@@ -43,6 +43,8 @@ BagLauncher::BagLauncher(ros::NodeHandle nh, BLOptions options):
     if(options.publish_name) {
         name_publisher_ = nh.advertise<bag_recorder::Rosbag>(sanitize_topic(options.name_topic), 5);
     }
+    // Set up the timer to call check_all every 0.5 secs
+    timer_ = nh.createTimer(ros::Duration(0.5), &BagLauncher::check_all, this);
```
Comment (rien88) [inline src/bag_launcher.cpp:47]: I think it needs to check at a rate faster than our fastest topic, with the while loop we had it set to 1000Hz, but I think 200Hz would do

**Comment by rien88** [comment]
Per slack discussion, decided not to integrate these changes.

**Pull request closed without merging**